stock_multi_warehouse_security/models/stock_move.py

```diff
@@ -0,0 +1,32 @@
+# Copyright (C) 2019 Akretion
+# Copyright 2022 Foodles (http://www.foodles.co).
+# @author Pierre Verkest <pierreverkest84@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    warehouse_id = fields.Many2one(index=True)
+
+    @api.model
+    def _get_move_default_warehouse(self, vals):
+        if vals.get("picking_type_id"):
+            pick_type = self.env["stock.picking.type"].browse(vals["picking_type_id"])
+            wh = pick_type.warehouse_id
+        else:
+            locations = self.env["stock.location"].browse(
+                [vals["location_id"], vals["location_dest_id"]]
+            )
+            whs = locations.mapped("warehouse_id")
+            wh = len(whs) <= 1 and whs or self.env["stock.warehouse"]
+        return wh
+
+    @api.model
+    def create(self, vals):
```

Comment:
Would be nice to transform it in `@api.model_create_multi` I guess

Reply:
not sure to see the benefit here, if developer create multiple moves, according the code of  _get_move_default_warehouse each moves could concern different warehouse based on provided fields picking_type_id, location_id and location_dest_id. Do you want me to create a cache on those possible keys and reuse it to avoid few requests ?

We should probably carry on this thread on #1558